spec/mail/message_spec.rb

```diff
@@ -1403,6 +1403,23 @@ def message_headers_should_match(message, other)
 
       describe "content-transfer-encoding" do
 
+        # https://www.ietf.org/rfc/rfc2046.txt
+        # No encoding other than "7bit", "8bit", or "binary" is permitted for
+        # the body of a "message/rfc822" entity.
+        it "rfc2046" do
+          mail = Mail.new
+          mail.body << Mail::Part.new.tap do |part|
+            part.header[:content_disposition] = 'attachment; filename="test.eml"'
+            part.content_type  = 'message/rfc822'
+            part.body          = 'a' * 999
+            part.body.encoding = 'UTF-8'
+          end
+
+          part_transfer_encoding = mail.to_s.scan(/Content-Transfer-Encoding: (.+)\r$/).last.first
+          puts part_transfer_encoding
+          expect(part_transfer_encoding).to match(/7bit|8bit|binary/)
```

Comment:
Can parse the encoded email and check the content transfer encoding header directly.